libcaf_test/caf/test/dsl.hpp

```diff
@@ -373,7 +366,66 @@ class expect_clause {
   caf::scheduler::test_coordinator& sched_;
   caf::strong_actor_ptr src_;
   caf::abstract_actor* dest_;
-  std::function<void ()> peek_;
+  std::function<void()> peek_;
+};
+
+template <>
+class expect_clause<void> {
+public:
+  expect_clause(caf::scheduler::test_coordinator& sched)
```

Comment:
Why not mark this constructor `explicit`?
```suggestion
  explicit expect_clause(caf::scheduler::test_coordinator& sched)
```